agent/consul/multilimiter/multilimiter.go

```diff
@@ -145,14 +179,37 @@ func (m *MultiLimiter) Allow(e LimitedEntity) bool {
 			config = prefixConfig
 		}
 	}
-
 	limiter := &Limiter{limiter: rate.NewLimiter(config.Rate, config.Burst)}
-	limiter.lastAccess.Store(now)
-	tree, _, _ := limiters.Insert(e.Key(), limiter)
-	m.limiters.Store(tree)
+	limiter.lastAccess.Store(unixNow)
+	m.limiterCh <- &limiterWithKey{l: limiter, k: e.Key(), t: now}
 	return limiter.limiter.Allow()
 }
 
+func (m *MultiLimiter) runStoreOnce(ctx context.Context, waiter *time.Ticker, txn *radix.Txn) *radix.Txn {
+	select {
+	case <-waiter.C:
+		if txn != nil {
+			tree := txn.Commit()
+			m.limiters.Store(tree)
+			limiters := m.limiters.Load()
+			txn = limiters.Txn()
```

Comment:
As `limiters` is only modified by this goroutine, I think we can omit the `Load`:

```suggestion
			txn = tree.Txn()
```